actioncable/lib/assets/javascripts/action_cable/connection.coffee

```diff
@@ -56,14 +56,17 @@ class ActionCable.Connection
   events:
     message: (event) ->
       {identifier, message, type} = JSON.parse(event.data)
+      ping = identifier == ActionCable.INTERNAL.identifiers.ping
 
       switch type
         when message_types.confirmation
           @consumer.subscriptions.notify(identifier, "connected")
+          @dispatchEvent("connected", message, JSON.parse(identifier).channel)
         when message_types.rejection
           @consumer.subscriptions.reject(identifier)
         else
           @consumer.subscriptions.notify(identifier, "received", message)
+          @dispatchEvent("received", message, JSON.parse(identifier).channel) unless ping
```

Comment:
Line 58 parses `event.data`, which results in `identifier` being either `"_ping"`, or a string JSON like `"{"channel":"RecordChannel"}"`. Therefore I parse it separately when it is not ping.